sources/Tools/DatasetHandler.py

```diff
@@ -0,0 +1,146 @@
+from Tools.Describer import Describer
+
+logging.getLogger().setLevel(logging.INFO)
+
+
+class DatasetHandler(Visualiser, ArgParser, Describer):
+    header: list
+    df: pd.DataFrame
+    np_df: np.ndarray
+    np_df_train: np.ndarray
+    analysed_header: np.array
+    argparse_file_name: str
+    default_dataset: str = os.path.join("data", "datasets", "dataset_train.csv")
+    default_header_file: str = os.path.join("sources", "Ressources", "analysed_header.npy")
+
+    """
+        Override methods
+    """
+
+    def _add_parser_args(self, parser):
+        super()._add_parser_args(parser)
+        parser.add_argument(
+            "-f",
+            "--csv_file",
+            help=f"Provide CSV dataset file - Using '{self.default_dataset}' as default file",
+            type=str,
+            default=self.default_dataset,
+        )
+
+    def _get_options(self):
+        self.argparse_file_name = self.args.csv_file
+        if self.argparse_file_name == self.default_dataset:
+            logging.info("Using default dataset CSV file")
+        if (
+            not os.path.exists(self.argparse_file_name)
+            or os.path.splitext(self.argparse_file_name)[1] != ".csv"
+        ):
+            logging.error("The file doesn't exist or is in the wrong format\nProvide a CSV file")
+            sys.exit(-1)
+
+    """
+        Private methods
+    """
+
+    @staticmethod
+    def _load_npy(file_name: str):
+        try:
+            return np.load(file_name, allow_pickle=True)
+        except Exception as e:
+            logging.error(e)
+            sys.exit(-1)
+
+    @staticmethod
+    def _save_npy(file_name: str, data):
+        try:
+            np.save(file_name, data, allow_pickle=True)
+        except Exception as e:
+            logging.error(e)
+            sys.exit(-1)
+
+    def _check_header(self):
+        """
+        The logisitc regression is all about data.
+        Well choose headers from a list of header in a dataset is the main data analysis part
+        this function is oriented for machine learning programs
+        """
+        self.load_header()
+        self.header = list(self.raw_data.columns.values)
+        if not all(h in self.header for h in self.analysed_header):
+            logging.error("CSV file header doesn't contain enough data to analyse the dataset")
+            sys.exit(-1)
+
+    def _get_csv_file(self):
+        logging.info(f"Reading dataset from file: {self.argparse_file_name}")
+        try:
+            self.raw_data = pd.read_csv(f"{os.path.abspath(self.argparse_file_name)}")
+            self.raw_data.fillna(0, inplace=True)
+        except Exception:
+            logging.error(f"Error while processing {self.argparse_file_name}")
+            sys.exit(-1)
+        self._check_header()
+
+    @staticmethod
+    def _normalise(dataset: pd.DataFrame):
+        for h in dataset.columns:
+            dataset[h] = dataset[h].apply(lambda x: x / max(dataset[h]))
```

Comment:
I dont know if this could lead to a ZeroDivisionError, maybe add an try: except: